### PR TITLE
adding docs about marking branches as protected on github

### PR DESCRIPTION
**Pull request by mhrivnak**
re #2043
https://pulp.plan.io/issues/2043

### CODE REVIEW

**Comment by mhrivnak** [comment]
I'm not sure if this is the best place for these docs, so let me know if you have another idea. I cruised through the build docs and didn't see an obviously good place.

File: docs/dev-guide/contributing/branching.rst [hunk 1]
```diff
@@ -33,6 +33,18 @@ Each ``x.y`` release will have one corresponding branch called ``x.y-dev``. For
 example, all work for the 2.7.z series of releases gets merged into ``2.7-dev``.
 
 
+Protected Branches
+------------------
+
+Within each repository, the master branch, any branch ending in ``-dev``, and any
```
Comment (bmbouter) [inline docs/dev-guide/contributing/branching.rst]: Maybe s/master/``master``/  ?

File: docs/dev-guide/contributing/branching.rst [hunk 2]
```diff
@@ -33,6 +33,18 @@ Each ``x.y`` release will have one corresponding branch called ``x.y-dev``. For
...
+branch ending in ``-release`` should be marked as "protected" on GitHub. There is
```
Comment (bmbouter) [inline docs/dev-guide/contributing/branching.rst]: Maybe link the word "protected" to the github page on protected branches?  https://help.github.com/articles/about-protected-branches/

File: docs/dev-guide/contributing/branching.rst [hunk 3]
```diff
@@ -33,6 +33,18 @@ Each ``x.y`` release will have one corresponding branch called ``x.y-dev``. For
...
+a script at
+`devel/scripts/protect-branches.py
+<https://github.com/pulp/devel/tree/master/scripts/protected-branches.py>`_
```
Comment (bmbouter) [inline docs/dev-guide/contributing/branching.rst]: I think an additional sentence on which protections we use is important. We disallow force-push and delete, but we don't do anything with required checks.

**Comment by bmbouter** [comment]
I put some suggestions on here. There is only 1 blocking comment about specifying which types of protections we use. I'm adding the LGTM since I don't think re-review is necessary.

**Comment by bmbouter** [comment]
I also built the docs locally and there were no errors.

**Comment by mhrivnak** [comment]
Before merging, I'm just waiting on an LGTM here: https://github.com/pulp/devel/pull/3

**Pull request merged**